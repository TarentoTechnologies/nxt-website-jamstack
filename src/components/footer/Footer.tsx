import { Link } from "gatsby";
import ReactHtmlParser from "react-html-parser";

import {
  dLinkLeft,
  dLinkRight,
  directlink,
  dlinkLeft,
  dlinkRight,
  floatRight,
  footer,
  footerDesc,
  footerEmail,
  footerHeading,
  footerRight,
  marginBtm,
  marginBtm1,
  nxtLogo,
  rights,
  tarentoLogo,
  topBar,
} from "./Footer.module.css";

interface FooterProps {
  data: any;
}

export const Footer = ({ data }: FooterProps) => {
  console.log(data);
  return (
    <div className={`container-fluid ${footer}`}>
      <div className={`row`}>
        {/* Column one */}
        <div className={`col-sm-12 col-md-12 col-xl-8`}>
          <div className={`d-block d-md-none ${topBar}`}></div>
          <div className={`d-block d-md-none ${tarentoLogo} ${footerRight}`}>
            <Link to={data?.PrimaryLogoLink} rel="canonical">
              <img
                src={data?.PrimaryLogo?.localFile?.url}
                alt={data?.PrimaryLogo?.alternativeText}
                title={data?.PrimaryLogo?.caption}
                width="187"
                height="43"
              />
            </Link>
          </div>
          <div
            className={`d-none d-md-block d-xl-none ${tarentoLogo} ${footerRight} ${floatRight}`}
          >
            <a href={data?.SecondaryLogoLink} rel="canonical" target="_blank">
              <img
                src={data?.SecondaryLogo?.localFile?.url}
                alt={data?.SecondaryLogo?.alternativeText}
                title={data?.SecondaryLogo?.caption}
                width="187"
                height="43"
              />
            </a>
          </div>
          <div className={`${nxtLogo}`}>
          <Link to={data?.PrimaryLogoLink} rel="canonical">
              <img
                src={data?.PrimaryLogo?.localFile?.url}
                alt={data?.PrimaryLogo?.alternativeText}
                title={data?.PrimaryLogo?.caption}
                width="72"
                height="72"
              />
            </Link>
          </div>

          <div className={`d-none d-md-block ${topBar}`}></div>
          <div className={`row ${marginBtm}`}>
            <div className={`col-xs-12 col-sm-12 col-md-6 ${footerDesc}`}>
              <p>
                {ReactHtmlParser(
                  data?.PrimaryDescription?.data?.childMarkdownRemark?.html
                )}
              </p>
              {/* <p className={`${footerHeading}`}>
                Reach us at{" "}
                <a
                  href="mailto:hello@tarento.com;"
                  className={`${footerEmail}`}
                  rel="canonical"
                >
                  {data?.mailId}
                </a>
              </p> */}
            </div>
            <div className="col-md-6 d-flex p-0">
              <div
                className={`${directlink} ${dlinkLeft} ${dLinkLeft} offset-xl-2`}
              >
                <p className={`${footerHeading}`}>
                  {data?.PrimaryLevelOneHeading}
                </p>
                <ul>
                  {data?.PrimaryLevelOneList.map((item: any, index: number) => (
                    <li key={item?.id}>
                      <Link to={item?.Link} rel="canonical">
                        {item?.Label}
                      </Link>
                    </li>
                  ))}
                </ul>
              </div>
              <div
                className={`${directlink} ${dlinkRight} ${dLinkRight} offset-xl-2`}
              >
                <p className={`${footerHeading}`}>
                  {data?.PrimaryLevelTwoHeading}
                </p>
                <ul>
                  {data?.PrimaryLevelTwoList?.map(
                    (item: any, index: number) => (
                      <li key={item?.id}>
                        <Link to={item?.Link} rel="canonical">
                          {item?.Label}
                        </Link>
                      </li>
                    )
                  )}
                </ul>
              </div>
            </div>
          </div>
        </div>

        {/* Column two */}
        <div className={`col-sm-12 col-md-12 col-xl-4`}>
          <div className={`d-none d-xl-block ${tarentoLogo} ${footerRight}`}>
            <a href={data?.SecondaryLogoLink} rel="canonical" target="_blank">
              <img
                src={data?.SecondaryLogo?.localFile?.url}
                alt={data?.SecondaryLogo?.alternativeText}
                title={data?.SecondaryLogo?.caption}
                width="187"
                height="43"
              />
            </a>
          </div>
          <div className={`${topBar} col-md-6 col-lg-12`}></div>
          <div className={`row`}>
            <div
              className={`col-xs-6 col-sm-12 col-md-12 col-lg-12 ${marginBtm1} d-flex ps-0`}
            >
              <div className={`${directlink} ${dlinkLeft} offset-xl-1`}>
                <p className={`${footerHeading}`}>
                  {data?.SecondaryLevelOneHeading}
                </p>
                <ul>
                  {data?.SecondaryLevelOneList?.map(
                    (item: any, index: number) => (
                      <li key={item?.id}>
                        <a href={item?.Link} rel="canonical">
                          {item?.Label}
                        </a>
                      </li>
                    )
                  )}
                </ul>
              </div>
              <div
                className={`${directlink} ${dlinkRight} offset-lg-1 offset-xl-2`}
              >
                <p className={`${footerHeading}`}>
                  {data?.SecondaryLevelOneHeading}
                </p>
                <ul>
                  {data?.SecondaryLevelTwoList?.map(
                    (item: any, index: number) => (
                      <li key={item?.id}>
                        <a href={item?.Link} rel="canonical">
                          {item?.Label}
                        </a>
                      </li>
                    )
                  )}
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div className={`col-xs-12 col-sm-12 col-md-12 col-lg-12`}>
          <p className={`${footerHeading} ${rights}`}>{data?.Copyright}</p>
        </div>
      </div>
    </div>
  );
};
