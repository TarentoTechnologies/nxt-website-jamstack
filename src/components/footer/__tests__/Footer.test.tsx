import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import TarentoLogo from "../../../../static/images/company-logo.svg";
import NXTlogo from "../../../../static/images/logo-inner.svg";
import { Footer } from "../Footer";

const footerData = {
  nxtLogoImg: NXTlogo,
  nxtLogoImgTitle: "NXT-logo",
  companyLogoImg: TarentoLogo,
  companyLogoImgTitle: "Tarento-logo",
  description:
    "NXT is the innovation wing of Tarento with primary focus on bringing new technologies, new perspectives and new ways of working into Tarento. We take pride in being unconventional with our approaches while retaining the passion with which we work.",
  mailId: "hello@tarento.com",
  levelOneHeading: "Direct Links",
  levelOneList: [
    { title: "Home", href: "nxt.html" },
    { title: "Technology", href: "technology.html" },
    { title: "Design", href: "design-process.html" },
    { title: "Innovation", href: "nxt-innovation.html" },
    { title: "Pricing", href: "pricing.html" },
  ],
  levelTwoHeading: "Tech & Innovations",
  levelTwoList: [
    { title: "Bolt - Data Platform", href: "data-platform.html" },
    { title: "Rain - Analytics", href: "rain.html" },
    { title: "Thor - Chatbot", href: "thor.html" },
    { title: "GovTech", href: "govtech.html" },
    { title: "Customer Experience", href: "customer-experience.html" },
  ],
  levelThreeHeading: "Direct Links",
  levelThreeList: [
    { title: "Home", href: "https://www.tarento.com/" },
    { title: "About us", href: "https://www.tarento.com/about/" },
    { title: "Services", href: "https://www.tarento.com/services/" },
    { title: "Careers", href: "https://www.tarento.com/careers/" },
  ],
  levelFourHeading: "Programmes",
  levelFourList: [
    { title: "Lead", href: "https://www.tarento.com/lead/" },
    { title: "Blixt", href: "https://www.tarento.com/blixt/" },
  ],
  rightsText: "All rights reserved Â© 2023 Tarento Technologies.",
};

describe("Test if the Footer component is working as expected", () => {
  it("<Footer /> matches snapshot", () => {
    const component = render(<Footer data-testid="footer" data={footerData} />);
    expect(component.container).toMatchSnapshot();
  });

  it("<Footer /> renders without exploding", () => {
    const component = render(<Footer data-testid="footer" data={footerData} />);
    expect(component.getByTestId("footer")).toBeInTheDocument();
  });
});
