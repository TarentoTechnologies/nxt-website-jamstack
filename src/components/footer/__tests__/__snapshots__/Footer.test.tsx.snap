// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the Footer component is working as expected <Footer /> matches snapshot 1`] = `
<div>
  <div
    class="container-fluid footer"
    data-testid="footer"
  >
    <div
      class="row"
    >
      <div
        class="col-sm-12 col-md-12 col-xl-8"
      >
        <div
          class="d-block d-md-none topBar"
        />
        <div
          class="d-block d-md-none tarentoLogo footerRight"
        >
          <a
            rel="canonical"
          >
            <img
              data-aos="fade-up"
              height="43"
              width="187"
            />
          </a>
        </div>
        <div
          class="d-none d-md-block d-xl-none tarentoLogo footerRight floatRight"
        >
          <a
            rel="canonical"
            target="_blank"
          >
            <img
              data-aos="fade-up"
              height="43"
              width="187"
            />
          </a>
        </div>
        <div
          class="nxtLogo"
        >
          <a
            rel="canonical"
          >
            <img
              data-aos="fade-up"
              height="72"
              width="72"
            />
          </a>
        </div>
        <div
          class="d-none d-md-block topBar"
        />
        <div
          class="row marginBtm"
        >
          <div
            class="col-xs-12 col-sm-12 col-md-6 footerDesc"
          >
            <p
              class="footerReachUs"
              data-aos="fade-up"
            />
          </div>
          <div
            class="col-md-6 d-flex p-0"
          >
            <div
              class="directlink dlinkLeft dLinkLeft offset-xl-2"
            >
              <p
                class="footerHeading"
                data-aos="fade-up"
              />
              <ul />
            </div>
            <div
              class="directlink dlinkRight dLinkRight offset-xl-2"
            >
              <p
                class="footerHeading"
                data-aos="fade-up"
              />
              <ul />
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-sm-12 col-md-12 col-xl-4"
      >
        <div
          class="d-none d-xl-block tarentoLogo footerRight"
        >
          <a
            rel="canonical"
            target="_blank"
          >
            <img
              data-aos="fade-up"
              height="43"
              width="187"
            />
          </a>
        </div>
        <div
          class="topBar col-md-6 col-lg-12"
        />
        <div
          class="row"
        >
          <div
            class="col-xs-6 col-sm-12 col-md-12 col-lg-12 marginBtm1 d-flex ps-0"
          >
            <div
              class="directlink dlinkLeft offset-xl-1"
            >
              <p
                class="footerHeading"
                data-aos="fade-up"
              />
              <ul />
            </div>
            <div
              class="directlink dlinkRight offset-lg-1 offset-xl-2"
            >
              <p
                class="footerHeading"
                data-aos="fade-up"
              />
              <ul />
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-xs-12 col-sm-12 col-md-12 col-lg-12"
      >
        <p
          class="footerHeading rights"
        />
      </div>
    </div>
  </div>
</div>
`;
